integration tests from `track/1.7` are failing with `RuntimeError: command not found: secret-get`

### Bug Description

the `test_build_and_deploy` integration test for `track/1.7` is failing when it's trying to deploy the istio charms where `istio-pilot` charm goes to error. This happens because the tests in `track/1.7` are deploying istio charms from `latest/edge`, keeping in mind `track/1.7` is being tested on Juju `2.9` and the istio charms in latest need juju `3.0` for the secrets feature.

To fix that we need to pin the istio charms to the version that is compatible with ckf 1.7.
For ckf 1.7, we used istio 1.16.

### To Reproduce

run the individual charm or bundle integration tests in `track/1.7`

### Environment

juju 2.9/stable
microk8s 1.24/stable

### Relevant Log Output

```shell
Run tox -e controller-integration
controller-integration: freeze> python -m pip freeze --all
controller-integration: pip==24.0,setuptools==69.1.0,wheel==0.42.0
controller-integration: commands[0]> tox -c charms/jupyter-controller -e integration
integration: install_deps /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller> python -I -m pip install -r requirements-integration.txt
integration: commands[0] /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller> pytest -v --tb native --asyncio-mode=auto /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller/tests/integration --log-cli-level=INFO -s
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.2.0, pluggy-1.0.0 -- /home/runner/work/notebook-operators/notebook-operators/.tox/integration/bin/python
cachedir: /home/runner/work/notebook-operators/notebook-operators/.tox/integration/.pytest_cache
rootdir: /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller, configfile: pyproject.toml
plugins: anyio-3.7.1, asyncio-0.20.3, operator-0.22.0
asyncio: mode=auto
collecting ... collected 2 items

tests/integration/test_charm.py::test_build_and_deploy 
-------------------------------- live log setup --------------------------------
INFO     pytest_operator.plugin:plugin.py:625 Adding model github-pr-753bb-microk8s:test-charm-ryvr on cloud microk8s
-------------------------------- live log call ---------------------------------
INFO     juju.model:model.py:1981 Deploying ch:amd64/focal/istio-pilot-813
INFO     juju.model:model.py:1981 Deploying ch:amd64/focal/istio-gateway-837
INFO     juju.model:model.py:2618 Waiting for model:
  istio-pilot/0 [allocating] waiting: installing agent
  istio-ingressgateway/0 [allocating] waiting: installing agent
FAILED
tests/integration/test_charm.py::test_prometheus_integration XFAIL (...)
------------------------------ live log teardown -------------------------------
INFO     pytest_operator.plugin:plugin.py:768 Model status:

Model            Controller                Cloud/Region        Version  SLA          Timestamp
test-charm-ryvr  github-pr-753bb-microk8s  microk8s/localhost  2.9.34   unsupported  13:04:44Z

App                   Version  Status   Scale  Charm          Channel      Rev  Address        Exposed  Message
istio-ingressgateway           waiting      1  istio-gateway  latest/edge  837  10.152.183.70  no       installing agent
istio-pilot                    waiting      1  istio-pilot    latest/edge  813  10.152.183.26  no       installing agent

Unit                     Workload  Agent  Address      Ports  Message
istio-ingressgateway/0*  waiting   idle   10.1.31.139         List of <ops.model.Relation istio-pilot:1> versions not found for apps: istio-pilot
istio-pilot/0*           error     idle   10.1.31.138         hook failed: "istio-pilot-relation-created"



INFO     pytest_operator.plugin:plugin.py:774 Juju error logs:

unit-istio-pilot-0: 13:04:44 ERROR unit.istio-pilot/0.juju-log istio-pilot:1: Uncaught exception while in charm code:
Traceback (most recent call last):
  File "./src/charm.py", line 1224, in <module>
    main(Operator)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/main.py", line 441, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/main.py", line 149, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/framework.py", line 344, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/framework.py", line 841, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/framework.py", line 930, in _reemit
    custom_handler(event)
  File "./src/charm.py", line 367, in reconcile
    self._reconcile_gateway()
  File "./src/charm.py", line 681, in _reconcile_gateway
    "port": self._gateway_port,
  File "./src/charm.py", line 521, in _gateway_port
    if self._use_https():
  File "./src/charm.py", line 930, in _use_https
    return self._use_https_with_tls_provider() or self._use_https_with_tls_secret()
  File "./src/charm.py", line 942, in _use_https_with_tls_secret
    secret = self.model.get_secret(label=TLS_SECRET_LABEL)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/model.py", line 272, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/model.py", line 3148, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-istio-pilot-0/charm/venv/ops/model.py", line 2785, in _run
    raise RuntimeError(f'command not found: {args[0]}')
RuntimeError: command not found: secret-get
unit-istio-pilot-0: 13:04:44 ERROR juju.worker.uniter.operation hook "istio-pilot-relation-created" (via hook dispatching script: dispatch) failed: exit status 1

INFO     pytest_operator.plugin:plugin.py:790 juju-crashdump finished [0]
INFO     pytest_operator.plugin:plugin.py:861 Resetting model test-charm-ryvr...
INFO     pytest_operator.plugin:plugin.py:850    Destroying applications istio-pilot
INFO     pytest_operator.plugin:plugin.py:850    Destroying applications istio-ingressgateway
INFO     pytest_operator.plugin:plugin.py:866 Not waiting on reset to complete.
INFO     pytest_operator.plugin:plugin.py:839 Forgetting main...


=================================== FAILURES ===================================
____________________________ test_build_and_deploy _____________________________
Traceback (most recent call last):
  File "/home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller/tests/integration/test_charm.py", line 39, in test_build_and_deploy
    await ops_test.model.wait_for_idle(
  File "/home/runner/work/notebook-operators/notebook-operators/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 2610, in wait_for_idle
    _raise_for_status(errors, "error")
  File "/home/runner/work/notebook-operators/notebook-operators/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 2522, in _raise_for_status
    raise error_type("{}{} in {}: {}".format(
juju.errors.JujuUnitError: Unit in error: istio-pilot/0
------------------------------ Captured log setup ------------------------------
INFO     pytest_operator.plugin:plugin.py:625 Adding model github-pr-753bb-microk8s:test-charm-ryvr on cloud microk8s
------------------------------ Captured log call -------------------------------
INFO     juju.model:model.py:1981 Deploying ch:amd64/focal/istio-pilot-813
INFO     juju.model:model.py:1981 Deploying ch:amd64/focal/istio-gateway-837
INFO     juju.model:model.py:2618 Waiting for model:
  istio-pilot/0 [allocating] waiting: installing agent
  istio-ingressgateway/0 [allocating] waiting: installing agent
=========================== short test summary info ============================
FAILED tests/integration/test_charm.py::test_build_and_deploy - juju.errors.JujuUnitError: Unit in error: istio-pilot/0
======================== 1 failed, 1 xfailed in 39.72s =========================
integration: exit 1 (40.66 seconds) /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller> pytest -v --tb native --asyncio-mode=auto /home/runner/work/notebook-operators/notebook-operators/charms/jupyter-controller/tests/integration --log-cli-level=INFO -s pid=15710
  integration: FAIL code 1 (64.65=setup[24.00]+cmd[40.66] seconds)
  evaluation failed :( (64.71 seconds)
controller-integration: exit 1 (64.83 seconds) /home/runner/work/notebook-operators/notebook-operators> tox -c charms/jupyter-controller -e integration pid=15059
  controller-integration: FAIL code 1 (65.74=setup[0.91]+cmd[64.83] seconds)
  evaluation failed :( (65.82 seconds)
Error: Process completed with exit code 1.
```


### Additional Context

_No response_

--- a/charms/jupyter-controller/tests/integration/test_charm.py
+++ b/charms/jupyter-controller/tests/integration/test_charm.py
@@ -15,6 +15,7 @@ from pytest_operator.plugin import OpsTest
 
 log = logging.getLogger(__name__)
 
+ISTIO_VERSION = "1.16/stable"
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 CHARM_NAME = METADATA["name"]
 
@@ -24,14 +25,14 @@ async def test_build_and_deploy(ops_test: OpsTest):
     """Test build and deploy."""
     await ops_test.model.deploy(
         "istio-pilot",
-        channel="latest/edge",
+        channel=ISTIO_VERSION,
         config={"default-gateway": "test-gateway"},
         trust=True,
     )
     await ops_test.model.deploy(
         "istio-gateway",
         application_name="istio-ingressgateway",
-        channel="latest/edge",
+        channel=ISTIO_VERSION,
         config={"kind": "ingress"},
         trust=True,
     )

--- a/charms/jupyter-controller/tests/integration/test_charm.py
+++ b/charms/jupyter-controller/tests/integration/test_charm.py
@@ -75,7 +75,7 @@ async def test_prometheus_integration(ops_test: OpsTest):
         "--trust",
         check=True,
     )
-    await ops_test.model.deploy(prometheus_scrape, channel="latest/beta", config=scrape_config)
+    await ops_test.model.deploy(prometheus_scrape, channel="latest/stable", config=scrape_config)
 
     await ops_test.model.add_relation(CHARM_NAME, prometheus_scrape)
     await ops_test.model.add_relation(

--- a/charms/jupyter-controller/tests/integration/test_charm.py
+++ b/charms/jupyter-controller/tests/integration/test_charm.py
@@ -65,8 +65,17 @@ async def test_prometheus_integration(ops_test: OpsTest):
     scrape_config = {"scrape_interval": "30s"}
 
     # Deploy and relate prometheus
-    await ops_test.model.deploy(prometheus, channel="latest/stable", trust=True)
-    await ops_test.model.deploy(prometheus_scrape, channel="latest/stable", config=scrape_config)
+    await ops_test.juju(
+        "deploy",
+        prometheus,
+        "--channel",
+        "latest/edge",
+        "--revision",
+        "137",
+        "--trust",
+        check=True,
+    )
+    await ops_test.model.deploy(prometheus_scrape, channel="latest/beta", config=scrape_config)
 
     await ops_test.model.add_relation(CHARM_NAME, prometheus_scrape)
     await ops_test.model.add_relation(

--- a/tests/test_charms.py
+++ b/tests/test_charms.py
@@ -28,7 +28,8 @@ UI_METADATA = yaml.safe_load(Path(f"{UI_PATH}/metadata.yaml").read_text())
 CONTROLLER_APP_NAME = CONTROLLER_METADATA["name"]
 UI_APP_NAME = UI_METADATA["name"]
 
-
+ISTIO_VERSION = "1.16/stable"
+KUBEFLOW_VERSION = "1.7/stable"
 INGRESSGATEWAY_NAME = "istio-ingressgateway"
 PROFILE_NAME = "kubeflow-user"
 
@@ -65,8 +66,6 @@ def dummy_resources_for_testing(lightkube_client):
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test, lightkube_client, dummy_resources_for_testing):
-    # FIXME: once tracks for all kubeflow components are released, we should point to
-    # those instead of latest/edge
     # Build jupyter-controller and jupyter-ui
     controller_charm = await ops_test.build_charm(CONTROLLER_PATH)
     ui_charm = await ops_test.build_charm(UI_PATH)
@@ -78,14 +77,14 @@ async def test_build_and_deploy(ops_test, lightkube_client, dummy_resources_for_
     # Deploy istio-operators first
     await ops_test.model.deploy(
         "istio-pilot",
-        channel="latest/edge",
+        channel=ISTIO_VERSION,
         config={"default-gateway": "test-gateway"},
         trust=True,
     )
     await ops_test.model.deploy(
         "istio-gateway",
         application_name=INGRESSGATEWAY_NAME,
-        channel="latest/edge",
+        channel=ISTIO_VERSION,
         trust=True,
         config={"kind": "ingress"},
     )
@@ -106,9 +105,9 @@ async def test_build_and_deploy(ops_test, lightkube_client, dummy_resources_for_
 
     # Deploy jupyter-controller, admission-webhook, kubeflow-profiles and kubeflow-dashboard
     await ops_test.model.deploy(controller_charm, resources={"oci-image": controller_image_path})
-    await ops_test.model.deploy("admission-webhook", channel="latest/edge", trust=True)
-    await ops_test.model.deploy("kubeflow-profiles", channel="latest/edge", trust=True)
-    await ops_test.model.deploy("kubeflow-dashboard", channel="latest/edge", trust=True)
+    await ops_test.model.deploy("admission-webhook", channel=KUBEFLOW_VERSION, trust=True)
+    await ops_test.model.deploy("kubeflow-profiles", channel=KUBEFLOW_VERSION, trust=True)
+    await ops_test.model.deploy("kubeflow-dashboard", channel=KUBEFLOW_VERSION, trust=True)
     await ops_test.model.add_relation("kubeflow-profiles", "kubeflow-dashboard")
 
     # Wait for everything to deploy

